source/expressions.tex

```diff
@@ -3197,10 +3197,12 @@
 int x = f<int>();               // error: no argument for second function parameter
 \end{codeblock}
 \end{example}
-If the function is an implicit object member
-function, the \keyword{this} parameter of the function\iref{expr.prim.this}
-is initialized with a pointer to the object of the call, converted
-as if by an explicit type conversion\iref{expr.cast}.
+If the function is an implicit object member function,
+the object for which it is invoked\iref{expr.prim.this}
+is the object which would be pointed to by the result
```

Comment:
Why "would"? There is nothing hypothetical here. The member function is invoked for a certain subobject (or possibly the entire object) of the object designated by the object expression.

Reply:
> Why "would"? There is nothing hypothetical here.

We don't do any code transformations or anything else to make this conversion «really exist».

Reply:
We do convert the object expression to the type of the base class, and the result of that transformation is the value of "this" inside the member function, in my view.